src/main/java/edu/kis/powp/jobs2d/drivers/visitor/DriverCounterVisitor.java

```diff
@@ -1,55 +1,92 @@
 package edu.kis.powp.jobs2d.drivers.visitor;
 
+import edu.kis.powp.jobs2d.Job2dDriver;
 import edu.kis.powp.jobs2d.drivers.DriverMacro;
 import edu.kis.powp.jobs2d.drivers.PreciseLoggerDriver;
 import edu.kis.powp.jobs2d.drivers.adapter.LineDriverAdapter;
 import edu.kis.powp.jobs2d.drivers.adapter.TrackedJob2dDriver;
 import edu.kis.powp.jobs2d.drivers.composite.DriverContainer;
 import edu.kis.powp.jobs2d.features.driverTransofrmation.TransformingDriver;
 
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
 public class DriverCounterVisitor implements DriverVisitor {
 
-    private int count = 0;
+    private int visitDriverContainerCounter = 0;
+    private int visitLineDriverAdapterCounter = 0;
+    private int visitTrackedJob2dDriverCounter = 0;
+    private int visitDriverMacroCounter = 0;
+    private int visitPreciseLoggerDriverCounter = 0;
+    private int visitTransformingDriverCounter = 0;
 
-    public int getCount() {
-        return count;
+    public int getVisitDriverContainerCounter() {
+        return visitDriverContainerCounter;
+    }
+    public int getVisitLineDriverAdapterCounter() {
+        return visitLineDriverAdapterCounter;
+    }
+    public int getVisitTrackedJob2dDriverCounter() {
+        return visitTrackedJob2dDriverCounter;
+    }
+    public int getVisitDriverMacroCounter() {
+        return visitDriverMacroCounter;
+    }
+    public int getVisitPreciseLoggerDriverCounter() {
+        return visitPreciseLoggerDriverCounter;
+    }
+    public int getVisitTransformingDriverCounter() {
+        return visitTransformingDriverCounter;
     }
 
     @Override
     public void visitDriverContainer(DriverContainer driverContainer) {
-        count += 1;
+        visitDriverContainerCounter += 1;
+        for (Job2dDriver child : driverContainer.getChildren()) {
+            try {
+                Method acceptor = child.getClass().getMethod("accept", DriverVisitor.class);
+                acceptor.invoke(child, this);
+            } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                throw new RuntimeException(e);
+            }
+        }
     }
 
     @Override
     public void visitLineDriverAdapter(LineDriverAdapter lineDriverAdapter) {
-        count += 1;
+        visitLineDriverAdapterCounter += 1;
     }
```

Comment:
This is not a stateless approach. The count of LineDriverAdapter drivers in LineDriverAdapter driver is 1! This means:
count = 1; or
 visitLineDriverAdapterCounter = 1;
The stateless approach is not obligatory, nevertheless, I preferred the version with count, because now we do not have a TOTAL count getter.

Reply:
done